pkg/querier/http.go

```diff
@@ -64,8 +64,16 @@ func NewQuerierAPI(cfg Config, querier Querier, limits *validation.Overrides, lo
 
 // RangeQueryHandler is a http.HandlerFunc for range queries.
 func (q *QuerierAPI) RangeQueryHandler(w http.ResponseWriter, r *http.Request) {
+	log, ctx := spanlogger.New(r.Context(), "query.RangeQuery")
+
+	userID, err := tenant.TenantID(ctx)
+	if err != nil {
+		level.Error(log).Log("msg", "couldn't fetch tenantID", "err", err)
+	}
```

Comment:
I don't think you want to continue if a tenant ID is missing.

The suggestion below copies the error handling from `TailHandler` in the same file

```suggestion
	userID, err := tenant.TenantID(ctx)
	if err != nil {
		level.Warn(logger).Log("msg", "error getting tenant id", "err", err)
		serverutil.WriteError(httpgrpc.Errorf(http.StatusBadRequest, err.Error()), w)
		return
	}
```

Reply:
Done, thanks.